### PR TITLE
[bnb] Small improvements on utils

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes a small typo in `bitsandbytes.py`, should address https://github.com/huggingface/blog/pull/463#discussion_r946067141 
I will have to test it first and mark it as ready for review!

### CODE REVIEW

**Comment by younesbelkada** [comment]
Can confirm the tests pass!

**Comment by stas00** [comment]
so will there always be just one module not to convert?

won't it be safer to have modules instead and work with the list?

**Comment by younesbelkada** [comment]
I have proposed a small refactoring that includes:

- checking the list of modules to not convert instead of a single value. 
- changing an error message as it confused some user. Check: https://github.com/TimDettmers/bitsandbytes/issues/10

**Comment by younesbelkada** [comment]
From https://github.com/huggingface/transformers/issues/18660 I also just added a commit to support having a custom list of the keys to ignore

File: src/transformers/modeling_utils.py
Comment (stas00) [inline src/transformers/modeling_utils.py]:
```suggestion
                Jukebox that has several heads in different places and not necessarily at the last position.
```

File: src/transformers/modeling_utils.py
Comment (stas00) [inline src/transformers/modeling_utils.py:2148]: as you added in the docs, the above comment is outdated wrt "last" - it doesn't have to be last (jukebox)

File: src/transformers/modeling_utils.py
Comment (stas00) [inline src/transformers/modeling_utils.py:1750]: I'd definitely wait for @sgugger before merging this one, as this is a core API.

I'm not sure `no_load_in_8bit_modules` is very intuitive.  

I'd use `load_in_8bit` prefix for all 3 arguments, and then have the post_fix differentiator:

```
load_in_8bit
load_in_8bit_threshold
load_in_8bit_skip_modules
```

Comment (stas00) [inline src/transformers/modeling_utils.py:1750]: I'd definitely wait for @sgugger before merging this one, as this is a core API.

I'm not sure `no_load_in_8bit_modules` is very intuitive.  

I'd use `load_in_8bit` prefix for all 3 arguments, and then have the post_fix differentiator:

```
load_in_8bit
load_in_8bit_threshold
load_in_8bit_skip_modules
```

**Review by stas00 — CHANGES_REQUESTED** [review]

**Comment by younesbelkada** [comment]
Thanks a lot @stas00 !
There is no rush at all for this PR, we can definitely wait for @sgugger to move forward with it

File: src/transformers/modeling_utils.py
Comment (sgugger) [reply src/transformers/modeling_utils.py:1750]: Agree with Stas on the naming.

File: src/transformers/modeling_utils.py
Comment (sgugger) [inline src/transformers/modeling_utils.py:1845]: Would it make more sense to have this be a class variable of `PreTrainedModel` (like the `no_split` variable used for big model inference)? I'm afraid the user won't know what to set this too and it looks like it's something we should automatically handle?

**Review by sgugger — APPROVED** [review]
Thanks for working on this, I left some comments.

File: src/transformers/modeling_utils.py
Comment (younesbelkada) [reply src/transformers/modeling_utils.py]: Thanks ! Should be fixed in 27b0ef07a0e43676c525f131bea421e9f3c9f1a0

File: src/transformers/modeling_utils.py
Comment (younesbelkada) [reply src/transformers/modeling_utils.py]: Fixed in 01a4c0cea9d2c2ad8fd3c85afbdb274510d91344 !

File: src/transformers/modeling_utils.py
Comment (younesbelkada) [reply src/transformers/modeling_utils.py]: Thanks! Fixed in 224b504fc08506e5b8cafa164c4b8bcad454b81e

File: src/transformers/modeling_utils.py
Comment (younesbelkada) [reply src/transformers/modeling_utils.py]: I don't have a strong opinion on that but this argument is optional because the function `get_keys_not_to_convert` should automatically take care of that except for some models like Jukebox where it is a bit trickier due to its architecture. 
In this case the user will just have to manually set which modules should be kept in their native precision and specify them in the `kwargs`, so I feel like it is a bit easier than having it as an argument of `PretrainedModel` because you would need to open a PR to add the feature.

**Comment by younesbelkada** [comment]
Can confirm the tests are passing with the proposed fixes! Would love to have a final round of review 💪  
cc @sgugger @stas00

File: src/transformers/modeling_utils.py
Comment (sgugger) [inline src/transformers/modeling_utils.py:1762]:
```suggestion
            load_in_8bit_skip_modules (`List[str]`, *optional*):
```

**Review by sgugger — APPROVED** [review]
Still good for me. I'll let @stas00 have a second look since merging is blocked by his change request.

**Review by stas00 — APPROVED** [review]
thank you for addressing the suggestions, @younesbelkada

**Comment by younesbelkada** [comment]
Can confirm the slow tests pass after rebasing on `main`, will merge once it's green! 🟢

**Pull request merged**
